src/dispatch/static/dispatch/src/components/GenaiAnalysisDisplay.vue

```diff
@@ -0,0 +1,78 @@
+<template>
+  <div>
+    <h2>GenAI Analysis</h2>
+    <div v-if="analysis && Object.keys(analysis).length > 0">
+      <div v-for="(value, key) in analysis" :key="key" class="analysis-item">
+        <h3>
+          <b>{{ key }}</b>
+        </h3>
+        <div v-if="isObject(value)">
+          <!-- Render each sub-value with formatText to handle links and code formatting -->
+          <div
+            v-for="(subValue, subKey) in value"
+            :key="subKey"
+            v-html="formatText(subValue)"
```

Comment:
Do we need `v-html` here, I believe this is a risk of XXS vulnerabilities with it.

Reply:
Perhaps something like:

```vue
<template>
  <span>
    <template v-for="(part, index) in formattedParts" :key="index">
      <a v-if="part.type === 'link'" :href="part.href" target="_blank">{{ part.text }}</a>
      <code v-else-if="part.type === 'code'">{{ part.text }}</code>
      <span v-else>{{ part.text }}</span>
    </template>
  </span>
</template>

<script setup lang="ts">
import { computed } from 'vue';

const props = defineProps<{
  text: string;
}>();

interface TextPart {
  type: 'text' | 'link' | 'code';
  text: string;
  href?: string;
}

const formattedParts = computed<TextPart[]>(() => {
  const parts: TextPart[] = [];
  let remainingText = props.text;

  while (remainingText.length > 0) {
    const linkMatch = remainingText.match(/<([^|]+)\|([^>]+)>/);
    const codeMatch = remainingText.match(/`([^`]+)`/);

    if (!linkMatch && !codeMatch) {
      parts.push({ type: 'text', text: remainingText });
      break;
    }

    const linkIndex = linkMatch ? linkMatch.index! : Infinity;
    const codeIndex = codeMatch ? codeMatch.index! : Infinity;

    if (linkIndex < codeIndex) {
      if (linkIndex > 0) {
        parts.push({ type: 'text', text: remainingText.slice(0, linkIndex) });
      }
      parts.push({ type: 'link', href: linkMatch![1], text: linkMatch![2] });
      remainingText = remainingText.slice(linkIndex + linkMatch![0].length);
    } else {
      if (codeIndex > 0) {
        parts.push({ type: 'text', text: remainingText.slice(0, codeIndex) });
      }
      parts.push({ type: 'code', text: codeMatch![1] });
      remainingText = remainingText.slice(codeIndex + codeMatch![0].length);
    }
  }

  return parts;
});
</script>

<style scoped>
a {
  color: #1a73e8;
  text-decoration: underline;
}

code {
  font-family: monospace;
  background-color: #f5f5f5;
  padding: 0.2em 0.4em;
  border-radius: 4px;
}
</style>
```

Which could then be used like:

```vue
      <div v-for="(value, key) in analysis" :key="key" class="analysis-item">
        <h3>
          <b>{{ key }}</b>
        </h3>
        <div v-if="isObject(value)">
          <FormattedText
            v-for="(subValue, subKey) in value"
            :key="subKey"
            :text="subValue"
          />
        </div>
        <FormattedText v-else :text="value" />
      </div>
    </div>
```

Reply:
Yup, you're right that its usage could expose us to XSS attacks. Let me find a way to sanitize the contents.